File: 1658-minimum-operations-to-reduce-x-to-zero/1658-minimum-operations-to-reduce-x-to-zero.cpp
class Solution {
public:
    int minOperations(vector<int>& nums, int x) {
        int n = size(nums), target = accumulate(begin(nums), end(nums), 0);
        if ((target -= x) == 0) 
            return n;
        int currentSum = 0, maxLength = 0, left = 0;
        for (int right = 0; right < n; ++right) {
            currentSum += nums[right];
            while (left <= right and currentSum > target) 
                currentSum -= nums[left++];
            if (currentSum == target)
                maxLength = max(maxLength, right - left + 1);
        }
        return maxLength ? n - maxLength : -1;
    }
};

File: 1658-minimum-operations-to-reduce-x-to-zero/1658-minimum-operations-to-reduce-x-to-zero.py
class Solution:
    def minOperations(self, nums: List[int], x: int) -> int:
        sumArr, n = 0, len(nums)
        for num in nums: 
            sumArr += num
        k, currentSum, longestSubArr, i = sumArr - x, 0, 0, 0
        if k == 0: return n
        for j in range(n):
            currentSum += nums[j]
            if currentSum > k:
                while currentSum > k and i <= j:
                    currentSum -= nums[i]
                    i += 1
            if currentSum == k:
                longestSubArr = max(longestSubArr, j - i + 1)
        if longestSubArr == 0: return -1
        return n - longestSubArr
        
            